src/zenml/cli/secret.py

```diff
@@ -166,8 +165,13 @@ def register_secret(
             "the secret name."
         )
 
-    if "name" in parsed_args:
```

Comment:
Nope that was actually removed by mistake, thanks for catching that. However I'm not sure why this limitation even exists tbh, seems like we could simply remove it and allow `name` as a key inside a secret?

Reply:
I think this is a legacy thing, because the secret name used to be kept inside the secret contents, alongside the other keys. After my scoping changes, this is no longer an issue, bu not sure about the backwards compatible code.

Reply:
Yes that makes sense, we can keep it until we officially remove support for unscoped secrets managers.
I just re-checked and remembered that I removed this line on purpose though, as the same check already happens 5 lines above and was just duplicated :)